src/main/java/core/basesyntax/SquareSupplier.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class SquareSupplier {
+    private Random random = new Random();
+    private RandomSize size = new RandomSize();
+    private ColorSupplier color = new ColorSupplier();
+
+    public Figure getRandomSquare() {
+        return new Square(size.getRandomSize(), color.getRandomColor());
+    }

```

Comment:
Creating a figure, don't pass expressions in the constructor. For better code readability, create separate variables and pass them on. For example:

int sizeValue = size.getRandomSize();
Color colorValue = color.getRandomColor();
return new Square(sizeValue, colorValue);